webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala

```diff
@@ -1830,6 +1832,46 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
 
     } yield CanDeleteListResponseADM(iri, canDelete)
 
+  /**
+   * Deletes all comments from child list node.
+   */
+  private def deleteListChildNodeCommentsADM(
+    iri: IRI,
+    featureFactoryConfig: FeatureFactoryConfig,
+    requestingUser: UserADM
+  ): Future[ListChildNodeCommentsDeleteResponseADM] =
+    for {
+      node <- listNodeGetADM(
+                nodeIri = iri,
+                shallow = false,
+                featureFactoryConfig = featureFactoryConfig,
+                requestingUser = KnoraSystemInstances.Users.SystemUser
+              )
+
+      isRootNode = node match {
+                     case Some(_: ListRootNodeADM)  => true
+                     case Some(_: ListChildNodeADM) => false
+                     case _                         => false
+                   }
+
+      projectIri <- getProjectIriFromNode(iri, featureFactoryConfig)
+      namedGraph <- getDataNamedGraph(projectIri, featureFactoryConfig)
+
+      sparqlQuery <- Future(
+                       org.knora.webapi.messages.twirl.queries.sparql.admin.txt
+                         .deleteListChildNodeComments(
+                           namedGraph = namedGraph,
+                           nodeIri = iri,
+                           isRootNode = isRootNode
+                         )
+                         .toString()
+                     )
+
+      _: SparqlUpdateResponse <- (storeManager ? SparqlUpdateRequest(sparqlQuery))
+                                   .mapTo[SparqlUpdateResponse]
+
+    } yield ListChildNodeCommentsDeleteResponseADM(iri, !isRootNode)
```

Comment:
If a node doesn't have a comment, this will return true, even though nothing has been deleted.

Reply:
As above.